File: 242. Valid Anagram: O(nlogn).py
# 242. Valid Anagram
# https://leetcode.com/problems/valid-anagram/description/

# Trivial algorithm if string has anagram using built-in sorting in O(s*log(s) + t*log(t)):
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # Sort each string in O(nlogn):
        s_sorted = ''.join(sorted(s))
        t_sorted = ''.join(sorted(t))

        # Is anagram if sorted strings are equal:
        return s_sorted == t_sorted

File: 242. Valid Anagram: O(nlogn).rs
// 242. Valid Anagram
// https://leetcode.com/problems/valid-anagram/description/

// Trivial algorithm if string has anagram using built-in sorting in O(s*log(s) + t*log(t)):
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        // Sort each string in O(nlogn):
        let mut s_sorted = s.chars().collect::<Vec<char>>();
        let mut t_sorted = t.chars().collect::<Vec<char>>();
        s_sorted.sort();
        t_sorted.sort();

        // Is an anagram if sorted strings are equal:
        s_sorted == t_sorted
    }
}
